src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1253,9 +1253,28 @@ public static bool IsPossibleTupleContext(this SyntaxTree syntaxTree, SyntaxToke
             return false;
         }
 
-        public static bool IsPatternContext(this SyntaxTree syntaxTree, SyntaxToken leftToken, int position)
+        public static bool IsAtStartOfPattern(this SyntaxTree syntaxTree, SyntaxToken leftToken, int position)
         {
             leftToken = leftToken.GetPreviousTokenIfTouchingWord(position);
+            var node = leftToken.Parent;
```

Comment:
i would remove this entirely and make things explicit in code wrt just using leftToken.  i.e. `leftToken.Parent.IsKind(ParenthesizedExpression)`.  `node` is too vague and doesn' treall convey what it's talking about later on in the code.